### PR TITLE
Use Resolved attribute instead of BackgroundDependencyLoader wherever possible

### PR DESCRIPTION
**Pull request by recapitalverb**

### CODE REVIEW

File: osu.Game/Online/API/APIAccess.cs [hunk 1]
```diff
@@ -21,7 +22,9 @@ namespace osu.Game.Online.API
 {
     public class APIAccess : Component, IAPIProvider
     {
-        private readonly OsuConfigManager config;
+        [Resolved]
```
Comment (smoogipoo) [inline osu.Game/Online/API/APIAccess.cs]: This is also passed in in the ctor, and I think I'd rather keep that for now.

File: osu.Game/Online/Chat/ExternalLinkOpener.cs [hunk 2]
```diff
@@ -13,15 +13,17 @@ namespace osu.Game.Online.Chat
 {
     public class ExternalLinkOpener : Component
     {
-        private GameHost host;
-        private DialogOverlay dialogOverlay;
+        [Resolved(CanBeNull = true)]
```
Comment (smoogipoo) [inline osu.Game/Online/Chat/ExternalLinkOpener.cs]: Null-permission can be removed

File: osu.Game/Overlays/Settings/Sections/General/LoginSettings.cs [hunk 3]
```diff
@@ -201,7 +201,9 @@ private class LoginForm : FillFlowContainer
             private TextBox username;
             private TextBox password;
             private ShakeContainer shakeSignIn;
-            private IAPIProvider api;
+
+            [Resolved]
```
Comment (smoogipoo) [inline osu.Game/Overlays/Settings/Sections/General/LoginSettings.cs]: Should be nullable

**Comment by bdach** [comment]
FWIW, this is another thing that could be enforced with a Roslyn analyser if that ever happens.

**Pull request merged**